Source/WEB_ROOT/Inventory/psQuery/psQuery.js

```diff
@@ -0,0 +1,272 @@
+    author: Jim Parsons
+    version: 0.1
+    description: insert, update, and delete records via AJAX 
+    installation: <script src="/scripts/psQuery/psQuery.js"></script>
+    
+    usage: 
+        var table_object = $psq(table);
+        --- creates a table_object: 'table' can be the PS table number, or the table name ... eg. 'students' or 1
+        --- example:
+                var students_table = $psq('students');
+
+        table_object.insert(fields_values, callback_function);
+        --- inserts a new record into the table; if callback is supplied, will return the record ID (or 0 on failure)
+        --- fields_values is an object with field name and value pairs ... eg, {last_name: 'Smith', first_name: 'John'} 
+        --- example:
+                $psq('u_vex_vehicles').insert({studentsdcid: 123456, license: 'ABC123', make: 'Toyota' }, function(retID) {
+                   if  (retID) alert('Record Inserted: ' + retID);
+                });
+                
+        table_object.update(id, fields_values, callback_function);
+        --- updates an existing record in the table; if callback is supplied, will return the record ID (or 0 on failure)
+        --- id is either the DCID of stock tables, or the ID of extended schema tables
+        --- fields_values is an object with field name and value pairs ... eg, {last_name: 'Smith', first_name: 'John'} 
+        --- example:
+                $psq('students').update(12345, {last_name: 'Smith', first_name: 'John'}, function(retID) {
+                   if  (retID) alert('Record Updated: ' + retID);
+                });                
+        
+        table_object.delete(id, callback_function);
+        --- deletes an existing record in the table; if callback is supplied, will return the record ID (or 0 on failure)
+        --- id is either the DCID of stock tables, or the ID of extended schema tables
+        --- example:
+                $psq('log').delete(123456, function(retID) {
+                   if  (retID) alert('Record Deleted: ' + retID);
+                });
+
+*/
+
+;(function(global, $j) {
+    
+    var psQuery = function(table) {
+        return new psQuery.init(table);
+    }
+    
+    var maxParams = 25;
+    
+    // is this Admin, Teacher or Guardian portal ?? (note: script breaks where URL contains "/guardian/" or "/teachers/" in Admin portal)
+    var isGuardian = window.location.href.includes("/guardian/");
+    var isTeachers = window.location.href.includes("/teachers/");
+    
+    // selected stock tables, enables named parameter ... can add to if needed
+    var stockTables = {
+        students: '001',
+        courses: '002',
+        sections: '003',
+        cc: '004',
+        users: '005', // formerly teachers
+        gen: '006',
+        log: '008',
+        prefs: '009',
+        terms: '013',
+        schools: '039',
+        calendar_day: '051',
+        fees: '054',
+        guardians: '055'
+    };
+    
+    psQuery.prototype = {

```

Comment:
You're doing a pile of cool things here. I would break it up into 3 or 4 smaller methods. It's a bit easy to get lost in a method this large. Smaller methods with really distinct method names like DetermineTableType() ValidateForTableOrFields() would make the expression of what you've done here more readable.